### ISSUE TITLE
[Bug] DeltaFormer forward: a key tile straddling the chunk boundary skips the causal mask and reads the launch's own uninitialized output rows

### ISSUE THREAD

**Issue by truong-v** [fla-org/flash-linear-attention#1104]
### Checklist

- [x] I have checked [FAQs](https://github.com/fla-org/flash-linear-attention/blob/main/FAQs.md) and existing issues for similar problems
- [x] Please report this bug in English to ensure wider understanding and support

### Describe the Bug

DeltaFormer's forward walks the sequence in chunks of `C`, sharing one `u` buffer per sequence: at each launch, rows `[0, T-C)` hold earlier chunks' results and rows `[T-C, T)` are still uninitialized `torch.empty_like` memory that this launch is about to write. `parallel_deltaformer_fwd_kernel` decides **once per `BLOCK_T` tile, from the tile's start offset `kv_i` alone**, whether the tile is "earlier chunks" (read `u`, no causal mask) or "the current chunk" (causal mask, no `u` read):

https://github.com/fla-org/flash-linear-attention/blob/2501ac83fd2c1223f3f0393563f2d58255df9b5d/fla/ops/deltaformer/parallel.py#L168-L192

`BLOCK_T` is autotuned over {64, 32}. When it does not divide `C`, one tile straddles the `T - C` boundary and the whole tile takes the "earlier chunks" branch:

1. the causal mask is never applied, so the current chunk's own key columns enter `rowmax`/`rowsum` unmasked — `lse`, `w` and `u` all come out wrong; and
2. the `u` load is bounded only by `T`, so it pulls in rows `[T-C, T)` — the launch's own not-yet-written output region, i.e. uninitialized memory. The answer then also changes with whatever garbage the allocator left there, so results are not reproducible run to run.

`C` is a public, documented, unvalidated keyword of `deltaformer_attn` (and of `ParallelDeltaformerFunction.apply`). Any `C` not a multiple of 64 can fire; any `C` not a multiple of 32 fires at every autotune choice. Nobody noticed because the default `C = 512` and every tested shape keep the boundary 64-aligned. The backward kernel in the same file already handles the straddle correctly (`parallel_deltaformer_bwd_kernel_u` uses `kv_i + BLOCK_T >= ...` with an elementwise mask), so this is the forward missing the same treatment.

### Steps to Reproduce the Bug

Compares `u` from `ParallelDeltaformerFunction.apply` (the stage `deltaformer_attn` wraps; flash-attn only consumes `u` afterwards) against the recurrence of `fla/ops/deltaformer/naive.py`, and runs each config twice with different values planted in freed allocator memory:

```python
import math
import torch
import fla.ops.deltaformer.parallel as dfp
from fla.ops.deltaformer.naive import tril_softmax

def naive_u(q, k, v, beta):
    # stage 1 of fla/ops/deltaformer/naive.py (which only returns o):
    # u[t] = v[t] - beta[t] * sum_{j<t} softmax(q[t] @ k[:t]^T)[j] * u[j]
    qf, kf, vf = (t.float().transpose(1, 2) for t in (q, k, v))
    betaf = beta.float().transpose(1, 2)
    probs = tril_softmax(qf @ kf.transpose(-1, -2) * (1.0 / math.sqrt(q.shape[-1])), strict=True)
    us = []
    for t in range(q.shape[1]):
        u_t = vf[:, :, t]
        if t > 0:
            u_prev = torch.stack(us, dim=-2)
            u_t = u_t - betaf[:, :, t, None] * (probs[:, :, t, :t, None] * u_prev).sum(-2)
        us.append(u_t)
    return torch.stack(us, dim=2).transpose(1, 2).contiguous()

def poison_allocator(value, numel=1 << 22):
    junk = torch.full((numel,), value, device='cuda')
    del junk
    torch.cuda.synchronize()

torch.manual_seed(42)
B, T, H, D = 1, 64, 2, 64
q, k, v = (torch.randn(B, T, H, D, device='cuda') for _ in range(3))
beta = torch.randn(B, T, H, device='cuda').sigmoid()

for C, label in [(16, 'C=16  (straddling: BLOCK_T in {64, 32} does not divide 16)'),
                 (32, 'C=32  (straddling when the autotuner picks BLOCK_T=64)'),
                 (64, 'C=64  (aligned control)')]:
    with torch.no_grad():
        poison_allocator(777.0)
        u1 = dfp.ParallelDeltaformerFunction.apply(q, k, v, beta, C, None)
        poison_allocator(-333.0)
        u2 = dfp.ParallelDeltaformerFunction.apply(q, k, v, beta, C, None)
    ref = naive_u(q, k, v, beta)
    print(f'{label}\n    max |u - naive_u| {(u1 - ref).abs().max().item():.6f}'
          f'    max |u_run1 - u_run2| across allocator poisons {(u1 - u2).abs().max().item():.6f}')
```

```
C=16  (straddling: BLOCK_T in {64, 32} does not divide 16)
    max |u - naive_u| 0.486892    max |u_run1 - u_run2| across allocator poisons 0.559144
C=32  (straddling when the autotuner picks BLOCK_T=64)
    max |u - naive_u| 0.000998    max |u_run1 - u_run2| across allocator poisons 0.000000
C=64  (aligned control)
    max |u - naive_u| 0.000873    max |u_run1 - u_run2| across allocator poisons 0.000000
```

`compute-sanitizer --tool initcheck` with `PYTORCH_NO_CUDA_MEMORY_CACHING=1` on the `C=16` call reports 1,024 uninitialized global reads, all at the `u` load:

```
========= Uninitialized __global__ memory read of size 16 bytes
=========     at parallel_deltaformer_fwd_kernel+0x1cd0 in fla/ops/deltaformer/parallel.py:191
...
========= ERROR SUMMARY: 1024 errors
```

### Expected Behavior

`u` matches the naive recurrence within fp32 noise for every chunk size (as the aligned control does), deterministically. PR with the fix and regression tests: #PR_NUMBER.

### Environment Information

1. Torch: 2.13.0+cu129
2. Triton: 3.7.1
3. Commit tested: 2501ac83 (current main), Python 3.12, NVIDIA L40S, driver CUDA 12.4
